lib/auth/auth.go

```diff
@@ -5384,6 +5388,98 @@ func (a *Server) CompareAndSwapHeadlessAuthentication(ctx context.Context, old,
 	return headlessAuthn, trace.Wrap(err)
 }
 
+// GetResourceUsage returns the usage data for resources which are limited on usage-based billing plans.
+func (a *Server) GetResourceUsage(ctx context.Context, req *proto.GetResourceUsageRequest) (*proto.GetResourceUsageResponse, error) {
+	features := modules.GetModules().Features()
+	if !features.IsUsageBasedBilling {
+		return &proto.GetResourceUsageResponse{}, nil
+	}
+	accessRequestUsage, err := a.getAccessRequestMonthlyUsage(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &proto.GetResourceUsageResponse{
+		AccessRequests: accessRequestUsage,
+	}, nil
+}
+
+// getAccessRequestMonthlyUsage returns the number of access requests that have been used this month.
+// Only requests that were both created and since the start of the current month are included in this number.
+func (a *Server) getAccessRequestMonthlyUsage(ctx context.Context) (*proto.AccessRequestUsage, error) {
+	features := modules.GetModules().Features()
+	monthlyLimit := features.AccessRequests.MonthlyRequestLimit
+
+	now := a.clock.Now().UTC()
+	monthStart := time.Date(now.Year(), now.Month(), 1, 0, 0, 0, 0, time.UTC)
+
+	created := make(map[string]struct{})
+	reviewed := make(map[string]struct{})
+
+	var results []apievents.AuditEvent
+	var startKey string
+	var err error
+	for {
+		results, startKey, err = a.SearchEvents(ctx, events.SearchEventsRequest{
+			From:       monthStart,
+			To:         now,
+			Limit:      apidefaults.DefaultChunkSize,
+			Order:      types.EventOrderAscending,
+			EventTypes: []string{events.AccessRequestCreateEvent, events.AccessRequestReviewEvent},
+			StartKey:   startKey,
+		})
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		for _, ev := range results {
+			ev, ok := ev.(*apievents.AccessRequestCreate)
+			if !ok {
+				return nil, trace.BadParameter("expected *AccessRequestCreate, but got %T", ev)
+			}
+			id := ev.RequestID
+			switch ev.GetType() {
+			case events.AccessRequestCreateEvent:
+				created[id] = struct{}{}
+			case events.AccessRequestReviewEvent:
+				if _, ok := created[id]; ok {
+					reviewed[id] = struct{}{}
+				}
+			default:
+				log.Warnf("Expected event type %q or %q, got %q", events.AccessRequestCreateEvent, events.AccessRequestReviewEvent, ev.GetType())
+			}
+		}
+		if startKey == "" {
+			break
+		}
+	}
+
+	return &proto.AccessRequestUsage{
+		MonthlyLimit: int32(monthlyLimit),
+		MonthlyUsed:  int32(len(reviewed)),
+	}, nil
+}
+
+// verifyAccessRequestMonthlyLimit checks whether the cluster has exceeded the monthly access request limit.
+// If so, it returns an error. This is only applicable on usage-based billing plans.
+func (a *Server) verifyAccessRequestMonthlyLimit(ctx context.Context) error {
+	f := modules.GetModules().Features()
+	if !f.IsUsageBasedBilling {
+		return nil // unlimited
+	}
+
+	const limitReachedMessage = "cluster has reached its monthly access request limit, please contact the cluster administrator"
```

Comment:
Can you use an active voice here?

Reply:
It is active voice as far as I understand (`cluster has reached` as opposed to `limit has been reached`). I opted to phrase this as "cluster has reached ..." rather than "you have reached ...", seeing as this is a cluster-wide limit.

Reply:
I think this should work (active voice plus action to "fix it"):

>Your Teleport instance has reached the monthly limit, and you're not able to create more requests. If you want to keep using access request consider updating to Teleport Enterprise <link>.

CC @roraback